src/Compilers/CSharp/Test/Emit3/Semantics/PrimaryConstructorTests.cs

```diff
@@ -83,26 +83,60 @@ public void LanguageVersion_02(string typeKeyword, string baseKeyword)
 
 " + baseKeyword + @" Base{}
 ";
+            // For class inheriting from class without parameter list, we get the new specific error
+            // For other combinations (struct from interface), we get the generic error
+            var isClassInheritingFromClass = typeKeyword == "class" && baseKeyword == "class";
+            var errorCode = isClassInheritingFromClass
+                ? ErrorCode.ERR_UnexpectedArgumentListInBaseTypeWithoutParameterList
+                : ErrorCode.ERR_UnexpectedArgumentList;
+
             var comp = CreateCompilation(src1, parseOptions: TestOptions.Regular11, options: TestOptions.ReleaseDll);
-            comp.VerifyDiagnostics(
-                // (3,5): error CS8861: Unexpected argument list.
-                // Base()
-                Diagnostic(ErrorCode.ERR_UnexpectedArgumentList, "()").WithLocation(3, 5)
-                );
+            if (isClassInheritingFromClass)
+            {
+                comp.VerifyDiagnostics(
+                    // (3,5): error CS9339: Cannot pass arguments to the base type without a parameter list on the type declaration. Consider adding an empty parameter list to 'Point'.
+                    // Base()
+                    Diagnostic(errorCode, "()").WithArguments("Point").WithLocation(3, 5));
+            }
+            else
+            {
+                comp.VerifyDiagnostics(
+                    // (3,5): error CS8861: Unexpected argument list.
+                    // Base()
+                    Diagnostic(errorCode, "()").WithLocation(3, 5));
```

Comment:
The comment two lines above is incorrect half of the time. This also applies below in this test

Reply:
Not sure i understand what yuo mean.  The code this is reporting an error on is:

```c#
struct Point : Base() { }
interface Base { }
```

And it is specifically reporting on the `()` in the base list.  It seems correct to say that this parameter list is unexpected.